api/app/controllers/charges_controller.rb

```diff
@@ -41,4 +44,19 @@ def charge_params
     params.require(:line_items)
     params.permit(:merchant_id, line_items: [[:amount, :currency, :name, :quantity, :description]])
   end
+
+  def validate(line_item:)
+    [:amount, :currency, :name, :quantity].each do |attribute|
+      unless line_item.key?(attribute)
+        raise ActionController::ParameterMissing.new "param is missing or the value is empty: #{attribute}"
+      end
+    end
```

Comment:
```suggestion
unless ([:amount, :currency, :name, :quantity] - line_item.keys).empty?
  raise ActionController::ParameterMissing.new "all of a,b,c,d needed" 
end
    end
```

You do miss out by not seeing which is missing, but in the former, you find out one by one 🤷‍♂